ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/UnrestrictedPrivacyController.java

```diff
@@ -0,0 +1,298 @@
+    final BytesValueRLPOutput rlpOutput = new BytesValueRLPOutput();
+    privateTransaction.writeTo(rlpOutput);
+    return rlpOutput.encoded().toBase64String();
+  }
+
+  @Override
+  public Transaction createPrivacyMarkerTransaction(
+      final String privateTransactionLookupId, final PrivateTransaction privateTransaction) {
+    return privateMarkerTransactionFactory.create(privateTransactionLookupId, privateTransaction);
+  }
+
+  @Override
+  public Transaction createPrivacyMarkerTransaction(
+      final String privateTransactionLookupId,
+      final PrivateTransaction privateTransaction,
+      final Address privacyPrecompileAddress) {
+    return privateMarkerTransactionFactory.create(
+        privateTransactionLookupId, privateTransaction, privacyPrecompileAddress);
+  }
+
+  @Override
+  public ValidationResult<TransactionInvalidReason> validatePrivateTransaction(
+      final PrivateTransaction privateTransaction, final String enclavePublicKey) {
+    final String privacyGroupId = privateTransaction.determinePrivacyGroupId().toBase64String();
+    return privateTransactionValidator.validate(
+        privateTransaction,
+        determineBesuNonce(privateTransaction.getSender(), privacyGroupId, enclavePublicKey),
+        true);
+  }
+
+  @Override
+  public Optional<ExecutedPrivateTransaction> findPrivateTransactionByPmtHash(
+      final Hash pmtHash, final String enclaveKey) {
+
+    final Optional<Transaction> transaction = blockchain.getTransactionByHash(pmtHash);
+
+    if (transaction.isEmpty()) {
+      return Optional.empty();
+    }
+
+    final TransactionLocation transactionLocation =
+        blockchain.getTransactionLocation(pmtHash).orElseThrow();
+
+    final BlockHeader blockHeader =
+        blockchain.getBlockHeader(transactionLocation.getBlockHash()).orElseThrow();
+
+    final BytesValueRLPInput input = new BytesValueRLPInput(transaction.get().getPayload(), false);
+
+    final PrivateTransaction privateTransaction = PrivateTransaction.readFrom(input);
+
+    final String internalPrivacyGroupId =
+        privateTransaction.determinePrivacyGroupId().toBase64String();
+
+    final ExecutedPrivateTransaction executedPrivateTransaction =
+        new ExecutedPrivateTransaction(
+            blockHeader.getHash(),
+            blockHeader.getNumber(),
+            pmtHash,
+            transactionLocation.getTransactionIndex(),
+            internalPrivacyGroupId,
+            privateTransaction);
+
+    return Optional.of(executedPrivateTransaction);
+  }
+
+  @Override
+  public ReceiveResponse retrieveTransaction(
+      final String enclaveKey, final String enclavePublicKey) {
+    throw new PrivacyConfigurationNotSupportedException(
+        "Method not supported for UnrestrictedPrivacy");
+  }
+
+  @Override
+  public PrivacyGroup createPrivacyGroup(
+      final List<String> addresses,
+      final String name,
+      final String description,
+      final String enclavePublicKey) {
```

Comment:
##Keys
The problem is that there's nothing to hold the enclavePublicKey to the private transaction when using unrestricted privacy. It literally just puts the string in the payload. Whereas with an enclave you've got some guarantees. It would be down to the plugin to provide these kinds of guarantees. But yes, we should add the check to help stop the user from making a mistake.

Reply:
### Off-Chain
The thing is that this doesn't actually do anything! Whereas with an enclave it's communicating to other members to tell them this `privacyGroupId` means to distribute to these `participants`. With unrestricted there is no distribution, everyone gets it :smile:

Reply:
### Why?
The reason we need it is that a bunch of the `priv` endpoints only support using a `privacyGroupId` so we need a consistent way of creating groups when no using flexible on-chain privacy groups. 

The reality is that as it stands the `privacyGroupId` represents two things; state storage and participants. With `unrestricted` it can only be used to represent state storage. As such it can be thought of as an id for state storage. 

So when we call createPrivacyGroupId what we are really saying/asking is what state storage should I be using?

Reply:
I'm agree with Stefan the PrivacyGroupId should represent the state of a certain group of members, not only a state storage. Then you can restrict if you are allowed or not to target a private transaction to the group depending of the membership.

Reply:
You're right, the `privateGroupId` should be storage and state, that wasn't clear.

I guess I wanted to clarify responsibilities:

- **Storage and state** - besu 
- **Interacting with state** - besu (i.e priv json endpoints should work)
- **membership** - plugin (i.e who is a member of a `privacyGroup`)
- **permissioning** - plugin (i.e the plugin should decide if you can/can't send a tx to that `privacyGroup`)

Reply:
I almost agree with your list. I'm not too sure about permissioning. At the moment, with restricted privacy, besu does it, based on membership. Besu asks the enclave or the contract whether a user, identified by it's enclave public key, should get access to a privacy groups data.  "Permission" to use a cryptographic key has to be given by the plugin.